ee/localserver/request-query.go

```diff
@@ -19,48 +21,89 @@ func (ls *localServer) requestQueryHanlderFunc(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	var body map[string]any
+	var body map[string]string
 	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
 		sendClientError(w, fmt.Sprintf("error unmarshaling request body: %s", err))
 		return
 	}
 
-	queryRaw, ok := body["query"]
-	if !ok {
+	query, ok := body["query"]
+	if !ok || query == "" {
 		sendClientError(w, "no query key found in request body json")
 		return
 	}
 
-	query, ok := queryRaw.(string)
-	if !ok {
-		sendClientError(w, fmt.Sprintf("query value not a string: %s", query))
+	var results []map[string]string
+	var err error
+	backoff.WaitFor(func() error {
+		results, err = ls.querier.Query(query)
+		return err
+	}, 1*time.Second, 250*time.Millisecond)
+
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error executing query: %s", err))
+		return
+	}
+
+	jsonBytes, err := json.Marshal(results)
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error marshalling results to json: %s", err))
```

Comment:
Optional -- maybe we'd want a `sendServerError` so we can return a 5xx instead of a 4xx in these cases where it's not a client-side error?